msrestazure/azure_active_directory.py

```diff
@@ -274,7 +274,7 @@ def refresh_session(self, session=None):
                 self._default_token_cache(self.token)
         else:
             self.set_token()
-        return self.signed_session(session)
+        return self.signed_session(session or oauth_session)
```

Comment:
This line is not accurate, should be `self.signed_session(session)` only as before, since `oauth_session` is only used to get the token and set the `self.token` attribute and nothing more.

Reply:
No problem, I'll update that in a little bit.